lib/rrule/rule.rb

```diff
@@ -94,39 +79,61 @@ def all_until(end_date: nil, limit: nil)
 
         results_with_time = generator.combine_dates_and_times(possible_days_of_year, timeset)
         results_with_time.sort.each do |this_result|
-          if end_date
-            if this_result > end_date
-              return result
-            end
-          end
-
           if options[:until]
             if this_result > options[:until]
-              return result
+              raise StopIteration
```

Comment:
I don't know enough about how to implement Enumerable. Do you need to `raise` here or can you just `return` or `break`? The docs say that `loop` will rescue `StopIteration`, but I assume the `loop` in this method rescues it.